src/transformers/models/graphormer/algos_graphormer.pyx

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) Microsoft Corporation and HuggingFace
+# Licensed under the MIT License.
+
+import cython
+
+cimport numpy
+
+from cython.parallel cimport parallel, prange
+
+import numpy as np
+
+
+# Reduce this number if matrices are too big for large graphs
+UNREACHABLE_NODE_DISTANCE = 510 
+
+def floyd_warshall(adjacency_matrix):
+
```

Comment:
Might be useful to have some comments on what this function (and the next ones) does.